File: js/keypresses.js
fullscreen = false
sizenum = true


$(document).bind('keypress', function(e){
   if(e.key == 'n') {
     tglSizeNum()
   }
});

$(document).bind('keypress', function(e){
   if(e.key == 'u') {
     up()
   }
});

$(document).bind('keypress', function(e){
   if(e.key == 'd') {
     down()
   }
});

$(document).bind('keypress', function(e){
   if(e.key == 'c') {
     fill("calibration")
   }
});

$(document).bind('keypress', function(e){
  b = document.getElementById('body')
   if(e.key == 'f') {
     if (!fullscreen) {fullscreen=true;bodyFullScreen();b.classList.add('nocursor');return}
     if (fullscreen) {fullscreen=false;document.exitFullscreen();b.classList.remove('nocursor');return}
   }
});

$(document).bind('keypress', function(e){
   if(e.key == 'r') {
     fill(cSize, true)
   }
});

$(document).bind('keypress', function(e){
   x = document.getElementById('sizeOf20')
   if(e.key == '+' && x.style.display == 'inline') {
     upPxSize()
   }
});

$(document).bind('keypress', function(e){
   x = document.getElementById('sizeOf20')
   if(e.key == '-' && x.style.display == 'inline') {
     downPxSize()
   }
});

$(document).bind('keypress', function(e){
  if(e.key == '@') {
    toggleSizeOf20()
  }
});

$(document).bind('keypress', function(e){
   if(e.key == 'm') {
     tglMode()
   }
});

$(document).bind('keypress', function(e){
   if(e.key == 's') {
     tglSingle()
   }
});

$(document).bind('keypress', function(e){
   if(e.key == '!') {
     toggleMirror()
   }
});

$(document).bind('keypress', function(e){
  if(e.key == 'l') {
    tglLine();
  }
});

$(document).bind('keypress', function(e){
   if(['1', '2', '3', '4', '5', '6', '7', '8', '9'].includes(e.key)) {
     if (e.key === '1') {fill(1)}
     if (e.key === '2') {fill(3)}
     if (e.key === '3') {fill(5)}
     if (e.key === '4') {fill(6)}
     if (e.key === '5') {fill(8)}
     if (e.key === '6') {fill(10)}
     if (e.key === '7') {fill(11)}
     if (e.key === '8') {fill(14)}
     if (e.key === '9') {fill(17)}
   }
});

File: lib/visiontest/keypresses.rb
calibrationmode = false
roomlengthmode = false

def changePxSize(num)
  lines = File.readlines('js/settings.js')
  size_line = lines[1].split(" ")
  size = size_line[2].to_i
  lines[1] = "  localStorage.pxsize = #{size + num}\n"
  File.write('js/settings.js', lines.join)
  VisionTest::FirefoxCtrl.driver.execute_script size + num > size ? 'upPxSize()' : 'downPxSize()'
end

def changeRoomLength(num)
  lines = File.readlines('js/settings.js')
  size_line = lines[7].split(" ")
  size = size_line[2].to_i
  lines[7] = "  localStorage.roomLength = #{size + num}\n"
  File.write('js/settings.js', lines.join)
  VisionTest::FirefoxCtrl.driver.execute_script size + num > size ? 'upRoomLength()' : 'downRoomLength()'
end

def toggleMirror()
  lines = File.readlines('js/settings.js')
  mirror_line = lines[4].split(" ")
  mirrored = mirror_line[2].to_i
  lines[4] = "  localStorage.mirrored = #{mirrored == 0 ? '1' : '0'}\n"
  File.write('js/settings.js', lines.join)
  VisionTest::FirefoxCtrl.driver.execute_script 'toggleMirror()' 
end

VisionTest::Serial.oncode '1B' do; VisionTest::FirefoxCtrl.driver.execute_script 'fill(cSize, true)' end
VisionTest::Serial.oncode '03' do; roomlengthmode ? changeRoomLength(1) : VisionTest::FirefoxCtrl.driver.execute_script('up()') end
VisionTest::Serial.oncode '04' do; roomlengthmode ? changeRoomLength(-1) : VisionTest::FirefoxCtrl.driver.execute_script('down()') end
VisionTest::Serial.oncode '17' do; VisionTest::FirefoxCtrl.driver.execute_script 'up()' end
VisionTest::Serial.oncode '1F' do; VisionTest::FirefoxCtrl.driver.execute_script 'down()' end
VisionTest::Serial.oncode '1C' do; VisionTest::FirefoxCtrl.driver.execute_script 'up()' end
VisionTest::Serial.oncode '1A' do; VisionTest::FirefoxCtrl.driver.execute_script 'down()' end
VisionTest::Serial.oncode '05' do; calibrationmode ? changePxSize(1) : VisionTest::FirefoxCtrl.driver.execute_script('up()') end
VisionTest::Serial.oncode '06' do; calibrationmode ? changePxSize(-1) : VisionTest::FirefoxCtrl.driver.execute_script('down()') end
VisionTest::Serial.oncode '07' do; VisionTest::FirefoxCtrl.driver.execute_script 'tglLine()' end
VisionTest::Serial.oncode '08' do; VisionTest::FirefoxCtrl.driver.execute_script 'tglSingle()' end
VisionTest::Serial.oncode '09' do; VisionTest::FirefoxCtrl.driver.execute_script 'tglSizeNum()' end
VisionTest::Serial.oncode '0A' do; VisionTest::FirefoxCtrl.driver.execute_script 'while (mode != "letter") {tglMode()}' end
VisionTest::Serial.oncode '0B' do; VisionTest::FirefoxCtrl.driver.execute_script 'while (mode != "number") {tglMode()}' end
VisionTest::Serial.oncode '0C' do; VisionTest::FirefoxCtrl.driver.execute_script 'while (mode != "symbol") {tglMode()}' end
VisionTest::Serial.oncode '0F' do; VisionTest::FirefoxCtrl.driver.execute_script 'while (mode != "e") {tglMode()}' end
VisionTest::Serial.oncode '41' do; VisionTest::FirefoxCtrl.driver.execute_script 'fill(1, false, true)' end
VisionTest::Serial.oncode '42' do; VisionTest::FirefoxCtrl.driver.execute_script 'fill(3, false, true)' end
VisionTest::Serial.oncode '43' do; VisionTest::FirefoxCtrl.driver.execute_script 'fill(5, false, true)' end
VisionTest::Serial.oncode '45' do; VisionTest::FirefoxCtrl.driver.execute_script 'fill(6, false, true)' end
VisionTest::Serial.oncode '46' do; VisionTest::FirefoxCtrl.driver.execute_script 'fill(8, false, true)' end
VisionTest::Serial.oncode '47' do; VisionTest::FirefoxCtrl.driver.execute_script 'fill(10, false, true)' end
VisionTest::Serial.oncode '49' do; VisionTest::FirefoxCtrl.driver.execute_script 'fill(11, false, true)' end
VisionTest::Serial.oncode '4A' do; VisionTest::FirefoxCtrl.driver.execute_script 'fill(14, false, true)' end
VisionTest::Serial.oncode '4B' do; VisionTest::FirefoxCtrl.driver.execute_script 'fill(17, false, true)' end
VisionTest::Serial.oncode '00' do; VisionTest::FirefoxCtrl.driver.execute_script 'screenSaver();' end
VisionTest::Serial.oncode '01' do; VisionTest::FirefoxCtrl.driver.execute_script 'screenSaver();' end
VisionTest::Serial.oncode '02' do; VisionTest::FirefoxCtrl.driver.execute_script 'screenSaver();' end

# Calibration

VisionTest::Serial.oncode '44' do
  if $mode == :unix && File.exist?("/media/#{ENV['USER']}/VT/vtcalib")
    VisionTest::FirefoxCtrl.driver.execute_script 'toggleSizeOf20();'
    calibrationmode = !calibrationmode
    VisionTest::FirefoxCtrl.driver.execute_script calibrationmode ? 'fill("calibration")' : 'fill(1, false, true)'
  end
end
VisionTest::Serial.oncode '48' do
  if $mode == :unix && File.exist?("/media/#{ENV['USER']}/VT/vtcalib")
    VisionTest::FirefoxCtrl.driver.execute_script 'toggleRoomLength();'
    roomlengthmode = !roomlengthmode
  end
end

VisionTest::Serial.oncode '4C' do; toggleMirror if $mode == :unix && File.exist?("/media/#{ENV['USER']}/VT/vtcalib") end

